### PR TITLE
build: prebuilt fixups

### PR DESCRIPTION
**Pull request by mattklein123**
1) Don't omit frame pointers
2) Compile symbols
3) Compile optimized version of boringssl (oops)

Fixes https://github.com/lyft/envoy/issues/898

### CODE REVIEW

**Comment by mattklein123** [comment]
@htuch @PiotrSikora

**Comment by PiotrSikora** [comment]
@mattklein123 maybe we should put all those common `CFLAGS` and `CXXFLAGS` in `ci/build_container/Makefile` instead?

**Comment by PiotrSikora** [comment]
@mattklein123 btw: for the CMake, there is also `RelWithDebInfo` (`-O2 -g -DNDEBUG`), which we might want to use instead of `Release` (`-O3 -DNDEBUG`) to keep the flags consistent across all dependencies.

FWIW, that's why I prefer "Bazel-all" approach to dependencies ;)

**Comment by mattklein123** [comment]
> maybe we should put all those common CFLAGS and CXXFLAGS in ci/build_container/Makefile instead

Yeah I thought about this, I guess there is no harm. Will do that.

> btw: for the CMake, there is also RelWithDebInfo (-O2 -g -DNDEBUG), which we might want to use instead of Release (-O3 -DNDEBUG) to keep the flags consistent across all dependencies.

OK will switch (though FYI it looks to be already setting -ggdb).

**Comment by PiotrSikora** [comment]
@mattklein123 yeah, I mentioned `RelWithDebInfo` mostly for context, it's fine if you want to leave `Release` instead.

**Comment by mattklein123** [comment]
@PiotrSikora updated

File: ci/build_container/build_recipes/boringssl.sh [hunk 1]
```diff
@@ -7,8 +7,10 @@ COMMIT=be2ee342d3781ddb954f91f8a7e660c6f59e87e5
 git clone https://boringssl.googlesource.com/boringssl
 cd boringssl
 git reset --hard $COMMIT
-cmake .
-make
+cmake -DCMAKE_CXX_FLAGS:STRING="${CXXFLAGS} ${CPPFLAGS}" \
+  -DCMAKE_C_FLAGS:STRING="${CFLAGS} ${CPPFLAGS}" \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo .
+make VERBOSE=1
```
Comment (PiotrSikora) [inline ci/build_container/build_recipes/boringssl.sh]: The same changes should be made to `googletest.sh`.

**Comment by htuch** [comment]
At some point, we can can consider also passing down these options from `repository.bzl`.  This will allow us to have the external deps reflect the intended built options. This would be useful when we eventually want to support MSAN, since the external deps also need to be built with MSAN (as well as lib[std]c++, but that's another story).

**Comment by mattklein123** [comment]
@PiotrSikora @htuch updated

**Pull request merged**